### PR TITLE
chore: obscure grpc error response

### PR DESCRIPTION
**Pull request by Cifko**
Description
---
Obscure errors returned by grpc on base node.  The flag for obscuring is coming from config (default is false).

### CODE REVIEW

**Comment by sdbondi** [comment]
Looks good, but I feel like it should be the other way around - you should opt in to sending error messages rather than opt out. maybe something like `grpc_enable_debug_responses`

File: applications/tari_base_node/src/grpc/base_node_grpc_server.rs [hunk 1]
```diff
@@ -104,8 +105,21 @@ impl BaseNodeGrpcServer {
...
     }
+
+    pub fn obscure_error_flag(&self) -> bool {
+        self.obscure_grpc_error
+    }
+}
+
+pub fn obscure_error(obscure: bool, status: Status) -> Status {
+    if obscure {
+        Status::unknown("Error has occurred. Details are obscured.")
```
Comment (stringhandler) [inline applications/tari_base_node/src/grpc/base_node_grpc_server.rs]: The status code (int) should still be sent, just not the message

**Review by stringhandler — COMMENTED** [review]
Happy, but the status code should still be propagated, only the message and details hidden

File: applications/tari_base_node/src/grpc/base_node_grpc_server.rs (hunk 1 above)
Comment (Cifko) [reply applications/tari_base_node/src/grpc/base_node_grpc_server.rs]: Done.

**Comment by Cifko** [comment]
> Looks good, but I feel like it should be the other way around - you should opt in to sending error messages rather than opt out. maybe something like `grpc_enable_debug_responses`

I rewrote it to `report_error` (default is false).

**Review by stringhandler — DISMISSED** [review]

**Review by stringhandler — APPROVED** [review]

**Pull request merged**